### PR TITLE
csi: use ceph-csi exposed csi config map API

### PR DESCRIPTION
**Pull request by iPraveenParihar**
This commit uses an csi config map API exposed by Ceph-CSI. 
This will reduce the efforts of keeping the configuration in sync with Ceph-CSI.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by travisn** [comment]
Let's make the API version updates from https://github.com/rook/rook/pull/13422#issuecomment-1864746771 and then continue with this PR.

File: pkg/operator/ceph/csi/cluster_config_test.go [hunk 1]
```diff
@@ -17,46 +17,86 @@ limitations under the License.
...
 	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
 	"github.com/stretchr/testify/assert"
+
+	cephcsi "github.com/ceph/ceph-csi/api/deploy/kubernetes"
 )
 
+func unmarshal(s string) ([]CSIClusterConfigEntry, error) {
+	var csiClusterConfigEntry []CSIClusterConfigEntry
+	if err := json.Unmarshal([]byte(s), &csiClusterConfigEntry); err != nil {
+		return csiClusterConfigEntry, fmt.Errorf("Error unmarshaling expected JSON: %s", err)
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/cluster_config_test.go:35]:
```suggestion
return csiClusterConfigEntry, err
```

File: pkg/operator/ceph/csi/cluster_config_test.go [hunk 2]
```diff
@@ -17,46 +17,86 @@ limitations under the License.
...
+	}
+
+	return csiClusterConfigEntry, nil
+}
+
+func JsonCompare(t *testing.T, exceptedJSON, actualJSON string) {
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/cluster_config_test.go:41]: `JsonCompare` to `compareJson`

File: pkg/operator/ceph/csi/cluster_config_test.go [hunk 3]
```diff
@@ -323,9 +376,9 @@ func TestUpdateCsiClusterConfig(t *testing.T) {
 	t.Run("test multus cluster", func(t *testing.T) {
 		s, err = updateCsiClusterConfig("[]", "rook-ceph-1", &csiClusterConfigEntryMultus)
 		assert.NoError(t, err)
-		assert.Equal(t, `[{"clusterID":"rook-ceph-1","monitors":["1.2.3.4:5000"],"namespace":"rook-ceph-1","rbd":{"netNamespaceFilePath":"/var/run/netns/rook-ceph-1","radosNamespace":"rook-ceph-1"}}]`, s)
+		// assert.Equal(t, `[{"clusterID":"rook-ceph-1","monitors":["1.2.3.4:5000"],"rbd":{"netNamespaceFilePath":"/var/run/netns/rook-ceph-1","radosNamespace":"rook-ceph-1"},"namespace":"rook-ceph-1"}]`, s)
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/cluster_config_test.go:379]: remove commented code

File: pkg/apis/go.mod [hunk 4]
```diff
@@ -7,13 +7,22 @@ require (
...
 	github.com/kube-object-storage/lib-bucket-provisioner v0.0.0-20221122204822-d1a8c34382f1
 	github.com/libopenstorage/secrets v0.0.0-20231011182615-5f4b25ceede1
-	github.com/openshift/api v0.0.0-20231204192004-bfea29e5e6c4
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.8.4
 	k8s.io/api v0.29.0
 	k8s.io/apimachinery v0.29.0
 )
 
+require (
```
[the remark this answers is not in the document]
Comment (iPraveenParihar) [reply pkg/apis/go.mod:16]: I tried doing it. But the modcheck CI test fails.

File: pkg/operator/ceph/csi/cluster_config_test.go (hunk 2 above)
Comment (Madhu-1) [reply pkg/operator/ceph/csi/cluster_config_test.go]: sorry i made a mistake in the comment it should be `compareJSON`

**Review by Madhu-1 — APPROVED** [review]

**Review by travisn — APPROVED** [review]

**Pull request merged**